emsdk.py

```diff
@@ -2486,7 +2486,7 @@ def set_active_tools(tools_to_activate, permanently_activate):
 
   # Construct a .bat script that will be invoked to set env. vars and PATH
   if WINDOWS:
-    env_string = construct_env(tools_to_activate, False)
+    env_string = construct_env(tools_to_activate, permanently_activate)
```

Comment:
Hmm, there are two issues with this:

1. `construct_env` is intended to create the `emsdk_env.bat` script that one can use to enter the Emsdk-enabled command prompt locally. It should never globally set environment variables.
2. Not a fault of this PR, but `construct_env` should actually remove the permanently_activate=true support. In current emsdk, it is always called with permanently_activate=False, to avoid a catastrophic bug where `%PROGRAMFILES%` etc. wildcards were getting expanded in PATH. (and also https://github.com/emscripten-core/emsdk/issues/4). The rule of thumb is that one should never be using the `SETX` command to set system PATH.

The right location to do this would be later in the code block starting from the comment
`# Apply environment variables to global all users section.`, looks like that is missing the relevant registrations.

Reply:
So, a big update here. While the code remains to be able to set user environment variables in windows it really only runs in two modes: current session locals or system wide environment variables. I removed the code that was trying to delete the variables and causing errors since it doesn't really do anything.

I removed setx and the option to permanently activate from construct_env, but it's still used as a fallback to set registry variables where it already had the 1024 check in place.

I fixed the PostMessage error that was happening by using SendMessageWithTimeout instead. It only sends the message if a registry value actually changed. I also confirmed that the message is being received by other applications.

The one caveat that still exists is that you need to run ``emsdk activate`` before running ``emsdk activate --global`` so that it will use the correct python version.